src/cfnlint/config.py

```diff
@@ -225,6 +225,17 @@ def comma_separated_arg(string):
     return string.split(',')
 
 
+def comma_separated_rule_names(string):
```

Comment:
Added this with the hope that it would simplify things for evaluation when it comes to rule evaluations. But since one of the 2 supports wildcards essentially it would still present a need to unpack the values that match the prefix in the case of an overlap.
eg.
```
ignore_rules: [E3]
mandatory_rules: [E3001]
```
In the above scenario we would compare them and would need to leave all other rules to ignore except for the E3001.

@kddejong has there been any consideration on expanding the prefixes into all the rules they represent? This would result in higher overhead from iteration perspective but should convert comparisons into being explicit.

Reply:
@yangas are you saying should we expand `E3` into the full rule numbers?  I could absolutely see the value in that.  

We have to make sure we account for scanning multiple templates in one execution of cfn-lint.  Since the ignore rule can be specified in each template.